h2/connection.py

```diff
@@ -1487,6 +1487,10 @@ def _receive_push_promise_frame(self, frame):
         """
         Receive a push-promise frame on the connection.
         """
+        # A client cannot push - RFC 7540 § 8.2
+        if not self.config.client_side:
+            raise ProtocolError("Received pushed stream from a client")
+
         if not self.local_settings.enable_push:
             raise ProtocolError("Received pushed stream")
```

Comment:
So, here's my question: is this worth it? I think, ordinarily, an invalid push would hit this code block here. That means that we'd still raise a `ProtocolError`. It also means that now, on push promise receipt, we have an extra `if` test that always fires in order to check for an error that we'll then immediately look for.

Should we consider whether it'd be better to put the new check into this `if` block? Or, even more aggressively, should we just implement #316? If we implement #316 then, definitionally, a server will be forbidden from setting `ENABLE_PUSH` to `1` (because we'll tell it not to), which means that all servers will automatically fall into the `if` block here.